src/legacy/ui/public/vis/editors/default/components/default_editor_agg.tsx

```diff
@@ -0,0 +1,287 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import {
+  EuiAccordion,
+  EuiToolTip,
+  EuiButtonIcon,
+  EuiText,
+  EuiTextColor,
+  EuiSpacer,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIconTip,
+  Color,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { AggConfig } from '../../../';
+import { DefaultEditorAggParams } from './default_editor_agg_params';
+import { DefaultEditorAggCommonProps } from './default_editor_agg_common_props';
+
+interface DefaultEditorAggProps extends DefaultEditorAggCommonProps {
+  agg: AggConfig;
+  aggIndex: number;
+  aggIsTooLow: boolean;
+  dragHandleProps: {} | null;
+  isDraggable: boolean;
+  isLastBucket: boolean;
+  isRemovable: boolean;
+}
+
+function DefaultEditorAgg({
+  agg,
+  aggIndex,
+  aggIsTooLow,
+  dragHandleProps,
+  formIsTouched,
+  groupName,
+  isDraggable,
+  isLastBucket,
+  isRemovable,
+  metricAggs,
+  lastParentPipelineAggTitle,
+  state,
+  onAggParamsChange,
+  onAggTypeChange,
+  onToggleEnableAgg,
+  removeAgg,
+  setTouched,
+  setValidity,
+}: DefaultEditorAggProps) {
+  const [isEditorOpen, setIsEditorOpen] = useState(agg.brandNew);
+  const [validState, setValidState] = useState(true);
+  const showDescription = !isEditorOpen && validState;
+  const showError = !isEditorOpen && !validState;
+  let disabledParams;
+  let aggError;
+  const isLastBucketAgg =
+    groupName === 'buckets' && lastParentPipelineAggTitle && isLastBucket && agg.type;
+
+  const SchemaComponent = agg.schema.editorComponent;
+
+  if (isLastBucketAgg) {
+    if (['date_histogram', 'histogram'].includes(agg.type.name)) {
+      disabledParams = ['min_doc_count'];
+    } else {
+      aggError = i18n.translate('common.ui.aggTypes.metrics.wrongLastBucketTypeErrorMessage', {
+        defaultMessage:
+          'Last bucket aggregation must be "Date Histogram" or "Histogram" when using "{type}" metric aggregation.',
+        values: { type: lastParentPipelineAggTitle },
+        description: 'Date Histogram and Histogram should not be translated',
+      });
+    }
+  }
+
+  useEffect(() => {
+    if (isLastBucketAgg && ['date_histogram', 'histogram'].includes(agg.type.name)) {
+      onAggParamsChange(
+        agg.params,
+        'min_doc_count',
+        // "histogram" agg has an editor for "min_doc_count" param, which accepts boolean
+        // "date_histogram" agg doesn't have an editor for "min_doc_count" param, it should be set as a numeric value
+        agg.type.name === 'histogram' ? true : 0
+      );
+    }
+  }, [lastParentPipelineAggTitle, isLastBucket, agg.type]);
+
+  // Returns a description of the aggregation, for display in the collapsed agg header
+  const getDescription = () => {
```

Comment:
Is there a reason for function here?
Could we simplify it?
`const aggDescription = agg.type &&agg.type.makeLabel ? agg.type.makeLabel(agg) : '' `